src/api/app/models/review.rb

```diff
@@ -17,6 +17,16 @@ class NotFoundError < APIException
   validates :reason, length: { maximum: 65534 }
 
   validate :check_initial, on: [:create]
+  validate :validate_recursive
+
+  belongs_to :assigned_from, class_name: 'Review', foreign_key: :review_id
+  has_one :assigned_to, class_name: 'Review', foreign_key: :review_id
```

Comment:
This is entirely up to you but maybe I would rename the foreign key to something more descriptive i.e. `assigned_from_review_id`